### PR TITLE
feat: add a packageVersion to options

### PR DESCRIPTION
**Pull request by Emeryao**
add an option called `packageVersion` to enable update the package version before publishing to npm

### CODE REVIEW

**Comment by dianjuar** [comment]
You read my mind!

That a feature that I was thinking 🤯

**Comment by dianjuar** [comment]
Ping @Emeryao
Are you still there ?

**Comment by Emeryao** [comment]
yeah
is there anything to improve for the PR

**Comment by dianjuar** [comment]
Yes I started a code review and I suggest some change, let me know what do you think.

Also you must update your PR with master because some commits were made.

---------------------------

Let me know if you have time to do those changes that I suggested, if not, I have a plan to merge your PR as it, I don't want to miss your contribution

**Comment by Emeryao** [comment]
ok @dianjuar
I'll update my PR branch and please review the codes and I'll try to fix that if any

**Comment by Emeryao** [comment]
updated the PR branch with master

**Comment by dianjuar** [comment]
@Emeryao I think you didn't get it...
I already put some comments in the PR that you made.

For example:
image

-------------------

**Comment by Emeryao** [comment]
sorry about that but I just can't find your reviews on the [tab of changes](https://github.com/bikecoders/ngx-deploy-npm/pull/10/files)  
have you finished them yet? @dianjuar

File: src/deploy/schema.json [hunk 1]
```diff
@@ -3,6 +3,10 @@
   "title": "schema",
```
Comment (dianjuar) [inline src/deploy/schema.json]: Use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.2/) on your commit message
`feat: add a packageVersion to options`

**Comment by dianjuar** [comment]
It's very estrange that you can't see the code review, I put it right away when you create the PR and I waited for your to start the discussion

**Comment by Emeryao** [comment]
> It's very estrange that you can't see the code review, I put it right away when you create the PR and I waited for your to start the discussion

yeah and I feel strange too
I even open this PR with the `incognito` mode of chrome and still no lucky

File: src/engine/engine.ts [hunk 2]
```diff
@@ -51,7 +61,7 @@ export function getOptionsString(options: Schema) {
   return (
     Object.keys(options)
       // Get only options with value
-      .filter(optKey => !!options[optKey])
+      .filter(optKey => !!options[optKey] && optKey !== 'packageVersion')
```
Comment (Emeryao) [inline src/engine/engine.ts]: use `!==` instead of `!=`

Comment (dianjuar) [inline src/engine/engine.ts]: This is braking the [Single Responsibility Use Principle](https://medium.com/@severinperez/writing-flexible-code-with-the-single-responsibility-principle-b71c4f3f883f). This function doesn't be aware about what our project needs, that filter needs to be done outside the function

**Comment by Emeryao** [comment]
Hi @dianjuar   
I'm just wondering if there is a `finish your reviews` button on the top right of the tab of changes?
there is one when I add some comment for review on the changes tab

File: src/engine/engine.ts [hunk 3]
```diff
@@ -12,20 +13,29 @@ export async function run(
...
+
     const { stdout, stderr } = await exec(commandToPublish);
 
     logger.info(stdout);
     logger.info(stderr);
 
     if (options.dryRun) {
-      await logger.info('The options are:');
-      await logger.info(JSON.stringify(options, null, 1));
+      logger.info('The options are:');
```
[the remark this answers is not in the document]
Comment (Emeryao) [reply src/engine/engine.ts]: finally the reviews :tada:

[the remark this answers is not in the document]
Comment (dianjuar) [reply src/engine/engine.ts]: WAS ME ALL THE TIME!!!!
Shame on me!
I did code reviews on other platforms but never in github so was not familiar with the process! Thank you @Emeryao for make me learn something new

**Comment by dianjuar** [comment]
> Hi @dianjuar
> I'm just wondering if there is a `finish your reviews` button on the top right of the tab of changes?
> there is one when I add some comment for review on the changes tab

Shame on me... Shame 🔔🔔🔔🔔

**Comment by dianjuar** [comment]
This is so embarrassing and awkward 😅. I did those comments like a month ago.

Let me know in the comments what do you think about them, remember that it's a discussion not an imposition.

**Comment by Emeryao** [comment]
I'm kinda busy this weekdays and I'm planning for these reviews this weekend

File: src/deploy/schema.json [hunk 4]
```diff
@@ -3,6 +3,10 @@
...
   "description": "Deployment/Publishment of Angular CLI applications to NPM",
   "properties": {
+    "packageVersion": {
+      "type": "string",
+      "description": "the version string of the publishing package"
```
[the remark this answers is not in the document]
Comment (Emeryao) [reply src/deploy/schema.json]: thining up descriptions is as hard as naming a variable so leave this to you

File: src/engine/engine.ts [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import { logging } from '@angular-devkit/core';
```
[the remark this answers is not in the document]
Comment (Emeryao) [reply src/engine/engine.ts]: please help since I'm not experienced

File: src/engine/engine.ts [hunk 6]
```diff
@@ -12,20 +13,29 @@ export async function run(
   try {
     options = exports.prepareOptions(options, logger);
 
-    const commandToPublish = `npm publish ${dir} ${exports.getOptionsString(
-      options
-    )}`;
+    if (options.packageVersion) {
+      let packageContent: string = fs.readFileSync(path.join(dir, 'package.json'), { encoding: 'utf8' });
```
[the remark this answers is not in the document]
Comment (Emeryao) [reply src/engine/engine.ts]: I think I have done it right please take a look

File: src/engine/engine.ts (hunk 2 above)
Comment (Emeryao) [reply src/engine/engine.ts]: I just delete `packgeVersion` from the options after using it  
and I do think there should be a function called `prepareOptionsForNpmCommand` to filter out the options that may cause a malfunction when passing to the command of `npm publish`

**Comment by Emeryao** [comment]
hi @dianjuar I think I have updated your reviews so please take a look again

---

and sorry for my poor English if it's confusing you

**Comment by dianjuar** [comment]
Thank you for your contribution :)

I will let you know once your contribution is published on NPM

**Pull request merged**